### DISCUSSION TITLE
prop spreading key in future react

### DISCUSSION THREAD

**Discussion by m-rutter** [TanStack/table#2356] [General]
So when I was migrating to typescript 3.9 I encountered various compiler error relating to the overwriting of key props (unreleated, but annoying), which brough to my attention the fact that react-table supplies a `key` prop in its most if not all of its get props functions. I believe that you might need to rethink this design as passing a key via props spreading might become depecrated in future versions of react:

https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md

**Comment by tannerlinsley** [comment]
Thanks for bringing that up. For now we know at the very least that with RT, you can always override the props, including the key, so it wouldn't need to be a breaking change. In the future, we'll definitely stop doing this and just supply a unique ID to use as the key instead (I am already doing this in the `next` branch)
